src/main/java/net/sf/jabref/gui/importer/ImportFormats.java

```diff
@@ -84,64 +59,56 @@ public static AbstractAction getImportAction(JabRefFrame frame, boolean openInNe
+                this.newDatabase = newDatabase;
+
+                if (newDatabase) {
+                    putValue(Action.NAME, Localization.menuTitle("Import into new database"));
+                    putValue(Action.ACCELERATOR_KEY, Globals.getKeyPrefs().getKey(KeyBinding.IMPORT_INTO_NEW_DATABASE));
+                } else {
+                    putValue(Action.NAME, Localization.menuTitle("Import into current database"));
+                    putValue(Action.ACCELERATOR_KEY, Globals.getKeyPrefs().getKey(KeyBinding.IMPORT_INTO_CURRENT_DATABASE));
+                }
             }
 
             @Override
             public void actionPerformed(ActionEvent e) {
-                JFileChooser fileChooser = createImportFileChooser(
-                        Globals.prefs.get(JabRefPreferences.IMPORT_WORKING_DIRECTORY));
-                int result = fileChooser.showOpenDialog(frame);
-
-                if (result != JFileChooser.APPROVE_OPTION) {
-                    return;
-                }
-
-                File file = fileChooser.getSelectedFile();
-                if (file == null) {
-                    return;
-                }
-
-                FileFilter ff = fileChooser.getFileFilter();
-                ImportFormat format = null;
-                if (ff instanceof ImportFileFilter) {
-                    format = ((ImportFileFilter) ff).getImportFormat();
-                }
-
-                try {
-                    if (!file.exists()) {
-                        // Warn that the file doesn't exists:
-                        JOptionPane.showMessageDialog(frame,
-                                Localization.lang("File not found") + ": '" + file.getName() + "'.",
-                                Localization.lang("Import"), JOptionPane.ERROR_MESSAGE);
-                        return;
+                SortedSet<ImportFormat> importers = Globals.IMPORT_FORMAT_READER.getImportFormats();
+                List<FileExtensions> extensions = importers.stream().map(p -> p.getExtensions()).collect(Collectors.toList());
+                FileDialog dialog = new FileDialog(frame, Globals.prefs.get(JabRefPreferences.IMPORT_WORKING_DIRECTORY));
+                // Add file filter for all supported types
+                List<String> flatExtensions = extensions.stream().flatMap(extList -> Stream.of(extList.getExtensions())).collect(Collectors.toList());
+                dialog.setFileFilter(new FileNameExtensionFilter(Localization.lang("Available import formats"), flatExtensions.toArray(new String[flatExtensions.size()])));
+                // Add filters for extensions
+                dialog.withExtensions(extensions);
+
+                Optional<Path> selectedFile = dialog.showDialogAndGetSelectedFile();
+
+                selectedFile.ifPresent(sel -> {
+                    try {
+                        File file = sel.toFile();
+
+                        if (!file.exists()) {
```

Comment:
I checked if there is a Path inside the Optional but not if the file exists. However, this will probably never happen as long as the file is not removed between the selection and the import. Was a artifact from the old code i was also thinking about...